### PR TITLE
Update to "Interface to Numpy"

### PR DESCRIPTION
**Pull request by VHchavez**
## Description
Address issue #2087. Specifically:

On `numpy.rst`:
  1) Updated Numpy quickstart guide link
  2) Updated Psi4Numpy Repo link
  3) Updated psi4.{Matrix, Vector, Dimension} -> psi4.core.{Matrix, Vector, Dimension}
  4) Added link to Psi4's Symmetry quick guide

On `numpy_helper.py`:
  1) Updated docstring example to `psi4.core.Matrix`. 

## Todos

- [ ]

## Questions
- [ ] Section "Array to Matrix" and "Matrix to Array" print and run the docstring. For "Matrix to Array" the example can be run with no problem. But "Matrix to Array" can only be run by its own class. Should this example be updated as well? e.g. replace both examples of
https://github.com/psi4/psi4/blob/master/psi4/driver/p4util/numpy_helper.py#L99 with:

```
  >>> data = np.random.rand(20,1)
  >>> vector = psi4.core.Matrix.from_array(data)
```
My concern is that the example would not be calling the function directly. What would be the best way to proceed?

## Checklist
- [ ] Tests added for any new features
- [ ] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

File: doc/sphinxman/source/numpy.rst [hunk 1]
```diff
@@ -101,13 +101,14 @@ this operation is identical to the above.
 |PSIfour| Data Objects with Irreps
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-|PSIfour| data objects natively support multiple irreducible representations
+|PSIfour| data objects natively support multiple `irreducible representations
+<https://psicode.org/psi4manual/4.0b3/psithoninput.html#symmetry>`_
```
Comment (loriab) [inline doc/sphinxman/source/numpy.rst:105]:
```suggestion
|PSIfour| data objects natively support multiple :ref:`irreducible representations
<sec:symmetry>`
```

**Review by loriab — COMMENTED** [review]
thanks for the updates.

yes, I think it's worth changing the docstring examples to run, even if `matrix_to_array` not called explicitly.

File: doc/sphinxman/source/numpy.rst (hunk 1 above)
Comment (jturney) [reply doc/sphinxman/source/numpy.rst:105]: I have learned a new GitHub trick: Suggested change   Nice!

**Review by jturney — APPROVED** [review]

**Comment by VHchavez** [comment]
Alright, I made the changes. I am now uncertain why Azure Pipelines was unhappy with one of the commits.

**Comment by loriab** [comment]
Thanks. We'll see how azure likes this state. It's a little tricky to access CI records for old commits.

**Review by loriab — APPROVED** [review]
lgtm! thanks for handling this

**Review by PeterKraus — APPROVED** [review]

**Pull request merged**